### DISCUSSION TITLE
Ability to set default expiration time for all Cache keys

### DISCUSSION THREAD

**Discussion by foremtehan** [laravel/framework#35374] [Ideas]
I'm working on a project which i found my self doing this : 

```php
Cache::put('thing', 'data', now()->addHours(5));

Cache::put('thing1', 'data', now()->addHours(5));

Cache::put('thing2', 'data', now()->addHours(5));

...
```

So it would be nice if we could set default expiration in `AppServiceProvider` for Cache facade, if the `$ttl` didnt provided use the default value.

**Comment by joelbutcher** [comment]
@foremtehan The default is 'indefinitely'

**Comment by foremtehan** [comment]
@joelbutcher Ok But I started this thread to gives a idea to manually set our default instead of "indefintely" 

Btw in new version i think this is 10 year not indefinitely

**Comment by foremtehan** [comment]
Something like:

```php
public function boot()
{
    Cache::setDefaultExpiration(now()->addHours(5));
}
```

@paras-malhotra what do you think ?

**Reply by paras-malhotra** [reply]
I'm not sure what the use case would be here. I believe most cache items would need specific TTLs (e.g. if you're caching a static homepage whose content may change daily, you might want to use 24 hours. Compare that with caching say user data queries, which would probably need a much lower TTL). Also, a default cache value may actually introduce problems such as [cache stampedes](https://en.wikipedia.org/wiki/Cache_stampede) when a lot of cache values expire at the exact same time.
